xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCC.cpp

```diff
@@ -164,22 +262,28 @@ DemuxPacket* CDVDDemuxCC::Decode()
       {
         int service = m_streamdata[i].service;
 
-        char *data;
-        int len;
+        std::string data;
         if (service == 0)
         {
           data = m_ccDecoder->m_cc608decoder->text;
-          len = m_ccDecoder->m_cc608decoder->textlen;
+          if (m_ccDecoder->m_cc608decoder->on_buf &&
+              m_ccDecoder->m_cc608decoder->on_buf->channel_no < CC_CHANNELS)
+          {
+            const int channelNo = m_ccDecoder->m_cc608decoder->on_buf->channel_no;
+            const int rowPos = m_ccDecoder->m_cc608decoder->on_buf->channel[channelNo].rowpos;
+            const cc_row_t currentRow =
+                m_ccDecoder->m_cc608decoder->on_buf->channel[channelNo].rows[rowPos];
+            ApplyStyleModifiers(data, currentRow);
+          }
```

Comment:
imo was better add this code and relative methods defined inside the decoder itself or put all in a new separate file to keep demuxer clean, because something similar will be also needed for CEA-708 (as you can see on my old PR #20469)

Reply:
That should be possible, I'll try to investigate